prow/plugins/lgtm/lgtm_test.go

```diff
@@ -1092,3 +1092,55 @@ func TestRemoveTreeHashComment(t *testing.T) {
 		t.Fatalf("expected deleted tree_hash comment but got none")
 	}
 }
+
+func TestHelpProvider(t *testing.T) {
+	cases := []struct {
+		name           string
+		config         *plugins.Configuration
+		enabledRepos   []string
+		err            bool
+		expectedConfig string
+	}{
+		{
+			name:         "Empty config",
+			config:       &plugins.Configuration{},
+			enabledRepos: []string{"org1", "org2/repo"},
+		},
+		{
+			name:         "Invalid enabledRepos",
+			config:       &plugins.Configuration{},
+			enabledRepos: []string{"org1", "org2/repo/extra"},
+			err:          true,
+		},
+		{
+			name: "All configs enabled",
+			config: &plugins.Configuration{
+				Lgtm: []plugins.Lgtm{
+					{
+						Repos:            []string{"org2"},
+						ReviewActsAsLgtm: true,
+						StoreTreeHash:    true,
+						StickyLgtmTeam:   "team1",
+					},
+				},
+			},
+			enabledRepos: []string{"org1", "org2/repo"},
+			expectedConfig: `The plugin has the following configuration:<ul>
```

Comment:
Validating the exact output seems brittle. It is lame if someone who improves this sentence in a few months has to go and update the test as well.

I'd be more inclined towards something more like
```golang
for _, msg := range tc.messages {
  if !strings.Contains(config, msg) { t.Errorf("Could not find %s in %s", msg, config) }
}
```
And then make this `messages: []string{"Squashing commits does not remove LGTM", "Reviews of "approve" or "request changes" act as adding or removing LGTM."}`

Although that then makes me want to put this in a function so I can do:
`messages: []string{squashMsg(true), reviewMessage(true), stickyTeam("team1")}`

Reply:
So `squashMsg(false)`, `reviewMessage(false)` and `stickyTeam("")` should return "" in order to ditch the tests when constructing the config message in lgtm.go?

Reply:
Or sorry, in that case I'd just do:

```golang
const sm = "Squashing commits does not remove LGTM"
def fm(s string) string {
  return fmt.Sprintf("Commits from "%s" do not remove LGTM.", s)
}
```

Then include/exclude squashMessage from the things we check depending on the scenario (might be good to have both includes: []string and excludes: []string to validate the message only appears when set)

Also you can't punt on that if you'd prefer.